src/requestHandlers/publish.ts

```diff
@@ -0,0 +1,228 @@
+import { Request, Response } from 'express';
+import { Sequelize } from 'sequelize';
+import {
+    assertArgumentsDefined, assertArgumentsNumber, sanitizeResponse, assertArgumentsPosition
+} from './utils';
+import { sequelize } from '../database';
+import { createPanel } from '../services/panelService';
+import { createHook } from '../services/hookService';
+import { Json } from 'sequelize/types/utils';
+import { _saveImageController, validateImageFile } from './image';
+import { _createPanelSetController } from './panelSet';
+
+// types 
+type hook = {position : Json, panel_index : number}
+type hookArray = Array<hook>;
+
+const _publishController = (sequelize : Sequelize) => async (
+    author_id: string,
+    panelImage1 : Express.Multer.File, panelImage2 : Express.Multer.File, panelImage3: Express.Multer.File, hooks : hookArray
+) => {
+
+    // make transaction
+    const t = await sequelize.transaction();
+    try {
+        let finalResponseObject  = '';
+
+        // make panel_set, call the controller as author validation is needed
+        const panel_set = await _createPanelSetController(sequelize, t)(author_id) as {author_id: string, id: number} | Error;
```

Comment:
Why not import `IPanelSet` from `../models` instead of recreating the interface manually?